ext/trace_source.c

```diff
@@ -0,0 +1,67 @@
+#include "trace_source.h"
+
+#include "configuration.h"
+#include "ddtrace.h"
+#include "priority_sampling/priority_sampling.h"
+#include "tracer_tag_propagation/tracer_tag_propagation.h"
+
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
+
+#define TRACE_SOURCE_APM (1 << 0)
+#define TRACE_SOURCE_ASM (1 << 1)
+#define TRACE_SOURCE_DSM (1 << 3)
+#define TRACE_SOURCE_DJM (1 << 7)
+#define TRACE_SOURCE_DBM (1 << 9)
+
+static zend_string *_dd_tag_p_ts;
+
+void ddtrace_trace_source_minit() {
+    _dd_tag_p_ts = zend_string_init_interned(ZEND_STRL(DD_P_TS_KEY), 1 /* permanent */);
+}
+
+void ddtrace_trace_source_rinit() {
+    DDTRACE_G(trace_source_bm) = 0;
+}
+
+zend_string *ddtrace_trace_source_get_encoded() {
+    // This should not happen, but just in case.
+    if (DDTRACE_G(trace_source_bm) > 0xFF) {
+        return NULL;
+    }
+    zend_string *encoded = zend_string_alloc(2, 0);
+    snprintf(ZSTR_VAL(encoded), 3, "%02x" PRIx64, DDTRACE_G(trace_source_bm));
+    return encoded;
+}
+
+bool ddtrace_trace_source_set_from_hexadecimal(zend_string *hexadecimal)
+{
+    if (!hexadecimal || ZSTR_LEN(hexadecimal) != 2) {
```

Comment:
This violates the `MUST allow for parsing masks of at least 32 bits to ensure forward compatibility` requirement.